lib/mongo/collection/view/aggregation.rb

```diff
@@ -136,6 +136,18 @@ def validate_collation!(server)
             raise Error::UnsupportedCollation.new
           end
         end
+
+        # Skip, sort, limit, projection are specified as pipeline stages
+        # rather than as options.
+        def cache_options
+          {
+            namespace: collection.namespace,
+            selector: pipeline,
```

Comment:
Is selector then a hash for queries and array for agg pipelines? Is this documented somewhere?

Reply:
Yes precisely! Not currently documented anywhere but I added a note to RUBY-2376.